packages/ui/Toggleable/Toggleable.js

```diff
@@ -133,145 +135,116 @@ const ToggleableHOC = hoc(defaultConfig, (config, Wrapped) => {
 	const {activate, deactivate, eventProps, prop, toggle, toggleProp} = config;
 	const defaultPropKey = 'default' + cap(prop);
 
-	return class Toggleable extends React.Component {
-		static propTypes = /** @lends ui/Toggleable.Toggleable.prototype */ {
-			/**
-			 * Default toggled state applied at construction when the toggled prop is `undefined` or
-			 * `null`.
-			 *
-			 * @name defaultSelected
-			 * @memberof ui/Toggleable.Toggleable.prototype
-			 * @type {Boolean}
-			 * @default false
-			 * @public
-			 */
-			[defaultPropKey]: PropTypes.bool,
-
-			/**
-			 * Whether or not the component is in a disabled state.
-			 *
-			 * @type {Boolean}
-			 * @public
-			 */
-			disabled: PropTypes.bool,
-
-			/**
-			 * Current toggled state.
-			 *
-			 * When set at construction, the component is considered 'controlled' and will only
-			 * update its internal value when updated by new props. If undefined, the component
-			 * is 'uncontrolled' and `Toggleable` will manage the toggled state using callbacks
-			 * defined by its configuration.
-			 *
-			 * @name selected
-			 * @memberof ui/Toggleable.Toggleable.prototype
-			 * @type {Boolean}
-			 * @public
-			 */
-			[prop]: PropTypes.bool,
-
-			/**
-			 * Event callback to notify that state should be toggled.
-			 *
-			 * @name onToggle
-			 * @memberof ui/Toggleable.Toggleable.prototype
-			 * @type {Function}
-			 * @public
-			 */
-			[toggle]: PropTypes.func
-		}
-
-		static defaultProps = {
-			disabled: false
-		}
-
-		constructor (props) {
-			super(props);
-
-			this.state = {
-				rendered: false,
-				active: null,
-				controlled: prop in props
+	const forwardWithEventProps = eventName => adaptEvent(
+		(ev, props) => ({...pick(eventProps, props), ...ev}),
+		forward(eventName)
+	);
+	const forwardActivate = forwardWithEventProps(activate);
+	const forwardDeactivate = forwardWithEventProps(deactivate);
+	const forwardToggle = forwardWithEventProps(toggle);
+	const forwardToggleProp = forwardWithEventProps(toggleProp);
+
+	function Toggleable (props) {
+		const updated = {...props};
+		const hook = useToggle({
+			disabled: props.disabled,
+			// FIX: Current behavior for Toggleable treats `null` as undefined so we coerce it here
```

Comment:
_Should_ this behavior change? In terms of something that toggles, I'd imagine that `false`, `null`, and `undefined` would all mean the same thing. What are your thoughts on this that didn't fit into this code-comment?

Reply:
I believe it should but can't within the confines of 3.x. We can discuss separately to what degree we want to maintain backward compatibility in order to deliver this work within the current major release.

Reply:
Let's definitely discuss soon. Something to think about between now and then, and something I'm really curious about, is how you'd approach an omission of a prop in JSX if this behavior were to change. For example
`<Check />` or `<Button selected={undefined} />`. Would this no longer be allowed?